File: SimpleFactory/SimpleFactory.py
import types


class SimpleFactory(object):
    @staticmethod
    def create(product_class_name, *a, **kw):
        clazz = globals().get(product_class_name)
        if clazz != None:
            return clazz(*a, **kw)
        raise RuntimeError("no class named: %s was found" 
            % product_class_name)


class Phone(object):
    def __init__(self, branch, os):
        self._branch = branch
        self._os = os

    def say(self):
        print("I am " + self._branch + 
            ", and my OS is: " + self._os)


class IPhone(Phone):
    def __init__(self, os):
        super(self.__class__, self).__init__("IPhone", os)


class Oneplus(Phone):
    def __init__(self, os):
        super(self.__class__, self).__init__("Oneplus", os)

if __name__ == "__main__":
    SimpleFactory.create("IPhone", "IOS8").say()
    SimpleFactory.create("Oneplus", "Android7").say()

File: SimpleFactory/SimpleFactory.java
import java.lang.reflect.Constructor;


abstract class Phone {
    private String branch;
    private String os;

    protected Phone(String branch, String os) {
        this.branch = branch;
        this.os = os;
    }

    public void say() {
        System.out.println("I am " + this.branch 
            + ", my OS is: " + this.os);
    }
}


class IPhone extends Phone {
    public IPhone(String os) {
        super("IPhone", os);
    }
}


class Oneplus extends Phone {
    public Oneplus(String os) {
        super("Oneplus", os);
    }
}


public class SimpleFactory {
    public static Phone create(String className, String os) {
        try {
            Class<?> clazz = Class.forName(className);
            Constructor<?> consutructor = 
                clazz.getDeclaredConstructor(String.class);
            Phone phone = (Phone) consutructor.newInstance(os);
            return phone;
        } catch (Exception ex) {
            throw new RuntimeException(ex.getMessage());
        }
    }

    public static void main(String[] args) {
        SimpleFactory.create("IPhone", "IOS8").say();
        SimpleFactory.create("Oneplus", "Android7").say();
    }
}
